src/components/Popups/TaskManager.jsx

```diff
@@ -1,204 +1,177 @@
-import PropTypes from 'prop-types';
-import { PureComponent } from 'react';
 import './style/Popup.scss';
+import { useState, useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { closeEdit } from '../../store/actions/modalAction';
+import { loadUserData } from '../../store/actions/userDataAction';
+import { saveData } from '../../store/actions/mainDataAction';
 import { Button } from '../Buttons/Button/Button';
 import { Input } from '../FormElements/Input';
 import { Textarea } from '../FormElements/Textarea';
 import { List } from '../FormElements/List';
-import { TASK_VALIDATIONS } from '../../services/constants';
-import {
-  getCurrentDate,
-  onFocusDate,
-  onBlurDate,
-  validateValues,
-  generateID,
-  convertDate,
-} from '../../services/helpers';
+import { TASK_VALIDATIONS, taskInit, taskInitVal } from '../../services/constants';
+import { onFocusDate, onBlurDate, validateValues, convertDate } from '../../services/helpers';
+import { addUserTasks } from '../../services/services';
 import { validateCategory, validateField } from '../../services/validation';
 
-export class TaskManager extends PureComponent {
-  constructor(props) {
-    super(props);
-    this.state = {
-      data: {
-        id: generateID(),
-        name: '',
-        description: '',
-        start: getCurrentDate(),
-        deadline: getCurrentDate(),
-        assigners: [],
-      },
-      validation: {
-        nameErr: false,
-        descriptionErr: false,
-        startErr: false,
-        deadlineErr: false,
-        textMessage: 'This field must have at least one character',
-        deadlineMessage: 'Date cannot be lower than current',
-      },
-    };
-  }
+export function TaskManager() {
+  const dispatch = useDispatch();
+  const [data, setData] = useState(taskInit);
+  const [validation, setValidation] = useState(taskInitVal);
+  let inputRef = useRef(null);
+  const {
+    main: { members, theme, tasks },
+    modal: { edit, selected },
+  } = useSelector((state) => state);
 
-  componentDidMount() {
-    const { tasks, selected, edit } = this.props;
+  useEffect(() => {
     if (edit) {
-      const { id, assigners, name, description, start, deadline } = tasks[selected];
-      this.setState({
-        data: {
-          id,
-          name,
-          description,
-          start,
-          deadline,
-          assigners,
-        },
-      });
+      setData(tasks[selected]);
     }
-  }
+  }, []);
 
-  validateData = (length) => {
-    const { data } = this.state;
-    const validation = validateCategory('tasks', data);
-    this.setState({ validation });
-    return validateValues(validation, length);
+  const close = () => {
+    return dispatch(closeEdit());
   };
```

Comment:
```suggestion
  const close = () =>  dispatch(closeEdit());
```

you can use arrow function syntax

Reply:
Thanks, replaced in all cases.